### ISSUE TITLE
Can anyone tell what's causing this behaviour?

### ISSUE THREAD

**Issue by jakre1234** [peco/peco#554]
Hi there, I have used peco in past and it worked completely fine with my linux box but when I tried it after a long time it seems to be causing some issues as shown below, is that an issue with the kali update or am I missing something?
Can anyone tell what is causing this behavior and how this can be fixed?
```
Os: Kali Linux 2023.3
Shell: Zsh
```
![pecoissue1](https://github.com/peco/peco/assets/89703702/b000edc1-7360-4744-afa7-9163e3893857)

**Comment by syohex** [comment]
- Could you reinstall peco and try again ? Do you still reproduce the issue ?
- Can you reproduce other shells such as bash ?

**Comment by jakre1234** [comment]
I tried both of these but the issue still persist, not sure though why. At first I thought it might be due to my terminal theme(Green on Black) so i tried switching back to Kali default dark terminal theme but no success :(

**Comment by syohex** [comment]
- Can you reproduce the issue with the other terminal applications ?
- Do you set wrong values to environment variables related to terminal like `TERM` something like that ? Please check `printenv` result.

**Comment by jakre1234** [comment]
These are the```printenv``` results: 
```
COLORTERM=truecolor
LESS_TERMCAP_se=
LESS_TERMCAP_so=
TERM=xterm-256color
LESS_TERMCAP_mb=
LESS_TERMCAP_me=
LESS_TERMCAP_md=
LESS_TERMCAP_ue=
LESS_TERMCAP_us=
```
Is the ```TERM=xterm-256color``` real culprit ? I see some issues regarding the 256color, not sure though

**Comment by syohex** [comment]
I'm not sure, what will happen if that value is changed, like `TERM=xterm`  or unset it ?

**Comment by dawidgorny** [comment]
I had the same issue on Debian linux. I spent some time on it, but couldn't figure out which set or unset env variable is causing that. Especially that this behavior started unexpectedly - perhaps after updating peco or other packages.

Finally I build from source and that fixed the issue.
https://github.com/peco/peco#building-peco-yourself

**Comment by jakre1234** [comment]
Yes correct, it started unexpectedly but I think it has to do with ```TERM``` environment variable, the default on Debian(kali Linux in my case) the TERM environment variable is set to ```xterm-256color```
As @syohex mentioned, exporting ```TERM=xterm``` solves the problem and peco works fine

**Comment by pawlosck** [comment]
I have the same problem. It happen after update Ubuntu 20.04 to 23.10 and next to 24.04. There is problem in Konsole (KDE), but at real console (Ctrl+Alt+Fx), everything is ok.

**Comment by e-rakavest** [comment]
The same behavior on Ubuntu 24.04
On Ubuntu 22.04 worked fine(

**Comment by e-rakavest** [comment]
Anything will be change? or we need create some workaround?

**Comment by e-rakavest** [comment]
I've resolve this issue and add  new own wrapper function  into fish  ~/.config/fish/functions  and use like npeco

`function npeco
  export TERM=xterm
  if test (count $argv) = 0
    set peco_flags --layout=bottom-up
  else
    set peco_flags --layout=bottom-up --query "$argv"
  end

  peco $peco_flags|read foo

  if [ $foo ]
    commandline $foo
  else
    commandline ''
  end
  export TERM=xterm-256color
end`

**Comment by skewll** [comment]
Peco has been broken on Ubuntu for a few years. The snap version worked on my recent setup. I'm having issues with code-server twrminal with thebversion from apt install peco. 

For systems is isnt working on, Here is the last working version:

peco version v0.5.3 (built with go1.10.4). I made a script to determine OS and ARCH to download the correct v0.5.3

'''
bash -c "$(curl -sL https://raw.githubusercontent.com/skewll/scripts/main/install_peco.sh)"

Or in my use case, docker exec

docker exec -it code-server bash -c "$(curl -sL https://raw.githubusercontent.com/skewll/scripts/main/install_peco.sh)"
"''

**Comment by skewll** [comment]
Peco has been broken on Ubuntu for a few years. The snap version worked on my recent setup. I'm having issues with code-server twrminal with thebversion from apt install peco. 

For systems is isnt working on, Here is the last working version:

peco version v0.5.3 (built with go1.10.4). I made a script to determine OS and ARCH to download the correct v0.5.3

'''
bash -c "$(curl -sL https://raw.githubusercontent.com/skewll/scripts/main/install_peco.sh)"

Or in my use case, docker exec

docker exec -it code-server bash -c "$(curl -sL https://raw.githubusercontent.com/skewll/scripts/main/install_peco.sh)"
"''
